File: project/src/nwr/player/RecallPos.java
/*
 *  "NorseWorld: Ragnarok", a roguelike game for PCs.
 *  Copyright (C) 2002-2008, 2014 by Serg V. Zhdanovskih (aka Alchemist).
 *
 *  This file is part of "NorseWorld: Ragnarok".
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package nwr.player;

import jzrlib.external.BinaryInputStream;
import jzrlib.external.BinaryOutputStream;
import jzrlib.core.FileVersion;
import jzrlib.core.Point;
import jzrlib.utils.StreamUtils;
import nwr.core.StaticData;
import java.io.IOException;

/**
 *
 * @author Serg V. Zhdanovskih
 */
public final class RecallPos extends MemoryEntry
{
    public int Layer;
    public Point Field;
    public Point Pos;

    public RecallPos(Object owner)
    {
        super(owner);
        this.Field = new Point();
        this.Pos = new Point();
    }

    @Override
    public String getDesc()
    {
        return "";
    }

    @Override
    public String getName()
    {
        return "";
    }

    @Override
    public byte getSerializeKind()
    {
        return StaticData.SID_RECALL_POS;
    }

    @Override
    public void loadFromStream(BinaryInputStream stream, FileVersion version) throws IOException
    {
        this.Layer = StreamUtils.readInt(stream);
        this.Field.X = StreamUtils.readInt(stream);
        this.Field.Y = StreamUtils.readInt(stream);
        this.Pos.X = StreamUtils.readInt(stream);
        this.Pos.Y = StreamUtils.readInt(stream);

        this.Sign = "RecallPos";
    }

    @Override
    public void saveToStream(BinaryOutputStream stream, FileVersion version) throws IOException
    {
        StreamUtils.writeInt(stream, this.Layer);
        StreamUtils.writeInt(stream, this.Field.X);
        StreamUtils.writeInt(stream, this.Field.Y);
        StreamUtils.writeInt(stream, this.Pos.X);
        StreamUtils.writeInt(stream, this.Pos.Y);
    }
}

File: project/Game/RecallPos.cs
/*
 *  "NorseWorld: Ragnarok", a roguelike game for PCs.
 *  Copyright (C) 2002-2008, 2014 by Serg V. Zhdanovskih (aka Alchemist).
 *
 *  this file is part of "NorseWorld: Ragnarok".
 *
 *  this program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  this program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

using System.IO;
using BSLib;
using ZRLib.Core;
using NWR.Core;

namespace NWR.Game
{
    public sealed class RecallPos : MemoryEntry
    {
        public int Layer;
        public ExtPoint Field;
        public ExtPoint Pos;

        public RecallPos(object owner)
            : base(owner)
        {
            Field = new ExtPoint();
            Pos = new ExtPoint();
        }

        public override string Desc
        {
            get {
                return "";
            }
        }

        public override string Name
        {
            get {
                return "";
            }
        }

        public override byte SerializeKind
        {
            get {
                return StaticData.SID_RECALL_POS;
            }
        }

        public override void LoadFromStream(BinaryReader  stream, FileVersion version)
        {
            Layer = StreamUtils.ReadInt(stream);
            Field.X = StreamUtils.ReadInt(stream);
            Field.Y = StreamUtils.ReadInt(stream);
            Pos.X = StreamUtils.ReadInt(stream);
            Pos.Y = StreamUtils.ReadInt(stream);

            Sign = "RecallPos";
        }

        public override void SaveToStream(BinaryWriter stream, FileVersion version)
        {
            StreamUtils.WriteInt(stream, Layer);
            StreamUtils.WriteInt(stream, Field.X);
            StreamUtils.WriteInt(stream, Field.Y);
            StreamUtils.WriteInt(stream, Pos.X);
            StreamUtils.WriteInt(stream, Pos.Y);
        }
    }
}
